auth/DERMethods.m

```diff
@@ -0,0 +1,106 @@
+classdef DERMethods
+    % Class DERMethods contains all methods pertaining to
+    % Distinguished Encoding Rules (DER) and converting them to naked
+    % formats without the DER headers.
+
+    properties (Constant, Access = protected)
+        DER_ECDSA256_PK_HEADER = uint8([48; 89; 48; 19; 6; 7; 42; 134; 72; 206; 61; 2; 1; 6; 8; 42; 134; 72; 206; ...
+                                        61; 3; 1; 7; 3; 66; 0; 4])
+    end
+
+    methods (Static, Access = public)
+
+        function public_key = DER2PK(der_key, key_type)
+            % strips DER key of all headers and outputs naked public
+            % key  [r; s]
+            switch key_type
```

Comment:
why make this a switch and not just if else

Reply:
I think a switch statement make more sense when branching among multiple outcomes. Most important, I'm telling any future programmers how best to expand this code (when ECDSA 384, 512, ECSchnorr 128, 192, 256 become necessary). For branching among multiple states, I think switch statements are easier to ready. It is also slightly more efficient.

Reply:
if thats the case should we make ECDSA256 the value of a property and then put that variable in as the case so that we can switch it easier